### ISSUE TITLE
Eci failures with FastRTPS

### ISSUE THREAD

**Issue by mjcarroll** [ros2/build_farmer#270]
Recently Eloquent builds with FastRTPS have been failing.

http://build.ros2.org/view/Eci/job/Eci__nightly-fastrtps-dynamic_ubuntu_bionic_amd64/

**Comment by cottsay** [comment]
I worked through showing that the following weren't the cause of the regression:
* Changes to any repositories in `ros2.repos`
* Changes to `ros_buildfarm`
* A new base image for `ubuntu:bionic`

The Jenkins logs were sufficient to demonstrate their innocence.

I was convinced that the problem was an update to the Bionic system packages, but after disabling updates and security repos, I was seeing the same error and getting less confident. We should probably find a way to make this more clear for future comparisons - maybe dumping the apt log to console or something.

After thinking through this critically, I decided to give a close look to vendor packages, and that's where I found the problem.

`foonathan_memory_vendor` [is cloning from master](https://github.com/eProsima/foonathan_memory_vendor/blob/ebf1fe58949148afd4352fbaddd9cf8d22b9e5a9/CMakeLists.txt#L57-L59). I recommend that we target a specific ref in all released ROS 2 distros right away - a push to the `foonathan/memory` repository could break source builds at any time.

[Here is a change](https://github.com/foonathan/memory/commit/293f88d3a7cc49b25ffd4e9f27b1e4a8e14ee0d7) to foonathan that we unknowingly took, which I believe is causing the break.
